terraform/modules/shared_vpc/main.tf

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+module "vpc" {
+  source       = "terraform-google-modules/network/google"
+  version      = "~> 5.0"
+  project_id   = var.project_id
+  network_name = var.vpc_network
+  routing_mode = "GLOBAL"
+
+  subnets = [
+    {
+      subnet_name               = var.serverless_subnet
+      subnet_ip                 = "10.10.10.0/28"
+      subnet_region             = var.region
+      subnet_private_access     = "true"
+      subnet_flow_logs          = "true"
+      subnet_flow_logs_interval = "INTERVAL_10_MIN"
+      subnet_flow_logs_sampling = 0.7
+      subnet_flow_logs_metadata = "INCLUDE_ALL_METADATA"
+    },
+    {
+      subnet_name   = "proxy-only-subnet"
+      subnet_ip     = "10.129.0.0/23"
+      subnet_region = var.region
+      purpose       = "REGIONAL_MANAGED_PROXY"
+      role          = "ACTIVE"
+    }
+  ]
+
+  firewall_rules = [
+    {
+      name                    = "serverless-to-vpc-connector"
+      description             = null
+      direction               = "INGRESS"
+      priority                = null
+      ranges                  = ["107.178.230.64/26", "35.199.224.0/19"]
```

Comment:
The range should be customizable, can you add them into vars?

Reply:
These are not variable